tracee/container_tracer.py

```diff
@@ -1026,33 +1052,34 @@ def print_event(self, cpu, data, size):
         except:
             return
 
-        if not self.json:
-            log.info("%-14f %-12d %-12d %-6d %-16s %-16s %-6d %-6d %-6d %-16d %s" % (
-                context.ts / 1000000.0, context.mnt_id, context.pid_id, context.uid,
-                eventname, comm, pid, tid, ppid, context.retval, " ".join(args)))
-        else:  # prepare data to be consumed by ultrabox
-            data = dict()
-            data["status"] = [0]
-            data["raw"] = ""
-            data["type"] = ["apicall"]
-            data["time"] = context.ts / 1000000.0
-            data["mnt_ns"] = context.mnt_id
-            data["pid_ns"] = context.pid_id
-            data["uid"] = context.uid
-            data["api"] = eventname
-            data["process_name"] = comm
-            data["pid"] = pid
-            data["tid"] = tid
-            data["ppid"] = ppid
-            data["return_value"] = context.retval
-            dict_args = dict()
-            args_len = len(args)
-            for i in range(args_len):
-                dict_args["p" + str(i)] = args[i].rstrip('\0')
-            data["arguments"] = dict_args
-
-            log.info(json.dumps(data))
-            self.events.append(data)
+        if eventname in self.events_to_trace:
```

Comment:
I don't think this is needed. If no kprobe attached for an event, we will never receive that event from the kernel

Reply:
This is to avoid tracing out the “essential_syscalls” (execve/execveat) if they aren’t selected